src/privacy_accounting/dp_query/dp_query.py

```diff
@@ -0,0 +1,114 @@
+import abc
+
+import torch
+
+import src.utils.torch_nest_utils as nest
+
+
+class DPQuery(object):
+    __metaclass__ = abc.ABCMeta
+
+    def set_ledger(self, ledger):
+        """ Sets the ledger for this query to record privacy events in.
+        :param ledger: A 'PrivacyLedger'.
+        """
+
+        del ledger
+        raise TypeError(
+            'DPQuery typ %s does not support set_ledger.' % type(self).__name__
+        )
+
+    def initial_global_state(self):
+        """ Initial state for the global settings of the DPQuery. """
+
+        return ()
```

Comment:
Again, just an interface, so any query should have its own initial global state, L2 clip and noise var for Gaussian mechanism

Reply:
You dont have to override this method. Because all of this type should return some kind of tuple, its better to just pass an empty tuple rather than require a downstream class to check for None